common/test/acceptance/tests/studio/test_studio_components.py

```diff
@@ -65,3 +66,120 @@ def test_add_advanced_problem(self, component):
         self.assertEqual(problem.name, component)
+
+
+class ComponentTest(ContainerBase):
+    """
+    Test class to add different components.
+    (Not the advanced components)
+    """
+    def setUp(self, is_staff=True):
+        """
+        Create a course with a section, subsection, and unit to which to add the component.
+        """
+        super(ComponentTest, self).setUp(is_staff=is_staff)
+        self.advanced_settings = AdvancedSettingsPage(
+            self.browser,
+            self.course_info['org'],
+            self.course_info['number'],
+            self.course_info['run']
+        )
+
+    def populate_course_fixture(self, course_fixture):
+        course_fixture.add_advanced_settings(
+            {u"advanced_modules": {"value": ["split_test"]}}
+        )
+
+        course_fixture.add_children(
+            XBlockFixtureDesc('chapter', 'Test Section').add_children(
+                XBlockFixtureDesc('sequential', 'Test Subsection').add_children(
+                    XBlockFixtureDesc('vertical', 'Test Unit')
+                )
+            )
+        )
+
+    def test_add_html_component(self):
+        """
+        Scenario: I can add HTML components
+        Given I am in Studio editing a new unit
+        When I add this type of HTML component:
+            | Component               |
+            | Text                    |
+            | Announcement            |
+            | Zooming Image Tool      |
+            | Raw HTML                |
+        Then I see HTML components in this order:
+            | Component               |
+            | Text                    |
+            | Announcement            |
+            | Zooming Image Tool      |
+            | Raw HTML                |
+        """
+        # Components to be added
+        components = ['Text', 'Announcement', 'Zooming Image Tool', 'Raw HTML']
+        self.go_to_unit_page()
+        container_page = ContainerPage(self.browser, None)
+        # Add components
+        add_components(container_page, 'html', components)
+        problems = [x_block.name for x_block in container_page.xblocks[1:]]
+        # Assert that components appear in same order as added.
+        self.assertEquals(problems, components)
+
+    def test_add_latex_html_component(self):
+        """
+        Scenario: I can add Latex HTML components
+        Given I am in Studio editing a new unit
+        Given I have enabled latex compiler
+        When I add this type of HTML component:
+            | Component               |
+            | E-text Written in LaTeX |
+        Then I see HTML components in this order:
+            | Component               |
+            | E-text Written in LaTeX |
+        """
+        # Latex component
+        component = 'E-text Written in LaTeX'
+        # Visit advanced settings page and enable latex compiler.
+        self.advanced_settings.visit()
+        self.assertTrue(self.advanced_settings.is_browser_on_page())
```

Comment:
visit() already [does a wait_for_page()](http://bok-choy.readthedocs.io/en/latest/_modules/bok_choy/page_object.html#PageObject.visit) which in turn [has a Promise for is_browser_on_page()](http://bok-choy.readthedocs.io/en/latest/_modules/bok_choy/page_object.html#PageObject.wait_for_page), so this assertion is not needed.